### DISCUSSION TITLE
Create ImageDimPlot from cell-by-gene and metadata table derived spatial object

### DISCUSSION THREAD

**Discussion by Penguinayee** [satijalab/seurat#6938] [Q&A]
Hi,
I tried to directly use the cell-by-gene and metadata table to analyze the spatial data from Vizgen's breast cancer data 
(https://info.vizgen.com/merscope-ffpe-solution) or
(https://info.vizgen.com/ffpe-showcase?submissionGuid=b446f6a7-0ec1-4f22-80cf-af01dba17df2).

With the following code I could finish cell clustering and plot UMAP

`#read cell by gene table
cell_gene_table <- read.csv("./HumanBreastCancerPatient1_cell_by_gene.csv", stringsAsFactors = F, header = T)
#Remove column 1
cell_gene_table <- cell_gene_table[, -1]

#remve "Blanks"
cell_gene_table <- cell_gene_table %>% select(-contains("Blank."))
cell_gene_table <- t(cell_gene_table)

#read metadata
cell_meta <- read.csv("./HumanBreastCancerPatient1_cell_metadata.csv", header = T)
cell_names <- cell_meta[,1]
cell_meta <- cell_meta[,-1]

#adding cell names
colnames(cell_gene_table) <- cell_names
corner(cell_gene_table)

#create seurat object
viz_seurat <- CreateSeuratObject(counts = cell_gene_table, meta.data = cell_meta)
View(viz_seurat@meta.data)

all.genes <- rownames(viz_seurat)
viz_seurat <- NormalizeData(object = viz_seurat, normalization.method = "CLR", margin = 2)
viz_seurat <- ScaleData(viz_seurat, features = all.genes)

viz_seurat<- RunPCA(viz_seurat, npcs = 30, features = rownames(viz_seurat))
viz_seurat<- RunUMAP(viz_seurat, dims = 1:30)
viz_seurat<- FindNeighbors(viz_seurat, reduction = "pca", dims = 1:30)
viz_seurat<- FindClusters(viz_seurat, resolution = 0.3)

DimPlot(viz_seurat, reduction = "umap")`

However, when I tried to create spatial projection by `ImageDimPlot(viz_seurat, cols = "polychrome", axes = TRUE)`, there is an error `stop("No compatible spatial coordinates present")`

Could anyone point out in the Seurat object, which field should I add the spatial information (center_x and center_y) to make ImageDimPlot possible?
![image](https://user-images.githubusercontent.com/49685944/217585527-8a444a48-a1fe-452d-9e29-4d64d4473844.png)

Thanks a lot!

**Comment by NIKO6666666** [comment]
I've encountered the same issue as you. Do you now know how to add the spatial coordinate information from meta.data to the images attribute?

**Comment by NIKO6666666** [comment]
cents <- CreateCentroids(vizgen.input$centroids)
segmentations.data <- list(
    "centroids" = cents,
    "segmentation" = NULL
  )
coords <- CreateFOV(
coords = cents, # Provide only the centroids object
type = "centroids", # Specify that only centroids are being used
molecules = NULL,
assay = "RNA"
)
vizgen.obj[["p2s2"]] <- coords
I can successfully run it following the above steps.
